### PR TITLE
Add unskippable updates

### PR DESCRIPTION
**Pull request by jkcdarunday**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -128,11 +131,24 @@ module.exports = class PearUpdater extends ReadyResource {
       fork: this.drive.core.fork
     }
 
+    this.snapshot = this.drive.checkout(checkout.length)
+
+    try {
+      const latestPackage = JSON.parse(await this.snapshot.get('/package.json'))
+      const unskippableUpdates = (latestPackage.pear?.stage?.unskippableUpdates || [])
+        .filter(u => u > old.length)
```
Comment (mafintosh) [inline index.js:139]: why is this under stage?

Comment (mafintosh) [reply index.js:139]: also needs to check the key, decode it with hypercore-id-encoding and b4a compare against our key

File: index.js [hunk 2]
```diff
@@ -128,11 +131,24 @@ module.exports = class PearUpdater extends ReadyResource {
...
+        .sort((a, b) => a - b)
+      if (unskippableUpdates.length > 0) {
+        this._updateTarget = unskippableUpdates[0]
+        checkout.length = this._updateTarget
+        this.snapshot.close()
```
Comment (mafintosh) [inline index.js:144]: await

File: index.js [hunk 3]
```diff
@@ -128,11 +131,24 @@ module.exports = class PearUpdater extends ReadyResource {
       fork: this.drive.core.fork
     }
 
+    this.snapshot = this.drive.checkout(checkout.length)
+
+    try {
+      const latestPackage = JSON.parse(await this.snapshot.get('/package.json'))
+      const unskippableUpdates = (latestPackage.pear?.stage?.unskippableUpdates || [])
+        .filter(u => u > old.length)
+        .sort((a, b) => a - b)
```
Comment (mafintosh) [inline index.js:140]: a.length - b.length

File: index.js [hunk 4]
```diff
@@ -61,6 +61,7 @@ module.exports = class PearUpdater extends ReadyResource {
     this.snapshot = null
     this.updated = false
     this.updating = false
+    this._updateTarget = null
```
Comment (mafintosh) [inline index.js:64]: why is this needed?

File: index.js (hunk 1 above)
Comment (jkcdarunday) [reply index.js]: Moved it to pear.platform.unskippableUpdates. Is this alright or should it be something like pear.updates.unskippable?

File: index.js (hunk 4 above)
Comment (jkcdarunday) [reply index.js:66]: This is so that it stops updating after it reaches the next closest unskippable update (until pear shuts down to apply the update by calling applyUpdate)

Tried using just checkout but after update() finishes and checkout is now the same as the next unskippable update, it would keep updating to the next unskippable update even without applyUpdate being called.

Comment (mafintosh) [reply index.js:66]: i see, sounds like we shoudl make a that a boolean flag instead

Comment (mafintosh) [reply index.js:66]: ie `.frozen`

Comment (mafintosh) [reply index.js:66]: which makes it not update again, you can also call it `.unskippable` boolean

File: index.js [hunk 5]
```diff
@@ -282,6 +306,8 @@ module.exports = class PearUpdater extends ReadyResource {
       await local.put('checkout', c.encode(checkout, { ...this.checkout, key: this.drive.core.key }))
       await local.close()
 
+      this.frozen = false
```
Comment (mafintosh) [inline index.js:309]: dont think we should ever unset this
